primary_monitor.go

```diff
@@ -0,0 +1,136 @@
+package litefs
+
+import (
+	"context"
+	"errors"
+	"sync"
+)
+
+var (
+	ErrNotReady = errors.New("awaiting first event")
+	ErrClosed   = errors.New("PrimaryMonitor closed")
+)
+
+// PrimaryMonitor monitors the current primary status of the LiteFS cluster.
+type PrimaryMonitor struct {
+	es    *EventSubscription
+	ready chan struct{}
+	m     sync.RWMutex
+
+	isPrimary bool
+	hostname  string
+	err       error
+}
```

Comment:
Should the `PrimaryMonitor` take a `Client` instead of an `EventSubscription` so that it can handle reconnects? You shouldn't have disconnects on a local node (unless LiteFS dies) but you could if you were monitoring a remote LiteFS node.

Reply:
The `EventSubscription` handles re-running the request if it terminates.